### ISSUE TITLE
Add an analyzer for Egress gateway without a NetworkPolicy

### ISSUE THREAD

**Issue by MorrisLaw** [istio/istio#52619]
**Describe the feature request**

In the Wiz 1337 Exploit there was the incorrect assumption that the sidecar, by itself, is an egress enforcement mechanism. This is an incorrect assumption. One way you can ensure that all traffic goes to the sidecar though is by using an Egress gateway. Specifically you must provide an Egress gateway WITH a NetworkPolicy in order to ensure that traffic flows through as expected. If you want to ensure that mesh traffic goes through the Gateway, the NetworkPolicy is absolutely crucial and is the only secure way to do so.

John Howard explains this in detail, but specifically how one could bypass the sidecar with incorrectly setup Egress policies . See his blog post from May 2023, [here](https://blog.howardjohn.info/posts/bypass-egress/). Here's a relevant snippet from that blog post:

> If you do need to enforce egress policies in Istio, the only secure way is to [use an Egress Gateway](https://istio.io/latest/docs/ops/best-practices/security/#securing-egress-traffic). This is secure because (and only if) we use NetworkPolicy in tandem, to ensure all traffic flows through the gateway. This differs from the sidecar, as the egress gateway is not in the same trust domain as the application.

**TLDR;** Update the analyzer to check for Egress gateways WITHOUT a NetworkPolicy, when users decide to add an Egress policy but forget that NetworkPolicy piece they are opening up themselves to bad assumptions and similar exploits.

**Describe alternatives you've considered**
n/a

**Affected product area (please put an X in all that apply)**
[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
